File: Src/Icm.Core/CommandLine/SubArgument.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;

namespace Icm.CommandLineTools
{
	/// <summary>
	///  Subargument of a command line option.
	/// </summary>
	/// <remarks></remarks>

	public class SubArgument
	{

		private readonly SubArgumentType _type;
		private readonly string _description;

		private readonly string _name;
		public string Name {
			get { return _name; }
		}

		public string Description {
			get { return _description; }
		}

		public SubArgumentType Type {
			get { return _type; }
		}

		private SubArgument(string sName, SubArgumentType eType) : this(sName, "", eType)
		{
		}

		private SubArgument(string sName, string sDescription, SubArgumentType eType)
		{
			_name = sName;
			_description = sDescription;
			_type = eType;
		}

		public static SubArgument List(string sName)
		{
			return new SubArgument(sName, SubArgumentType.List);
		}

		public static SubArgument Optional(string sName)
		{
			return new SubArgument(sName, SubArgumentType.Optional);
		}

		public static SubArgument Required(string sName)
		{
			return new SubArgument(sName, SubArgumentType.Required);
		}

		public static SubArgument List(string sName, string sDescription)
		{
			return new SubArgument(sName, sDescription, SubArgumentType.List);
		}

		public static SubArgument Optional(string sName, string sDescription)
		{
			return new SubArgument(sName, sDescription, SubArgumentType.Optional);
		}

		public static SubArgument Required(string sName, string sDescription)
		{
			return new SubArgument(sName, sDescription, SubArgumentType.Required);
		}

		public bool IsList()
		{
			return Type == SubArgumentType.List;
		}

		public bool IsOptional()
		{
			return Type == SubArgumentType.Optional;
		}

		public bool IsRequired()
		{
			return Type == SubArgumentType.Required;
		}
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================

File: Src2/Icm.Core/CommandLine/SubArgument.vb

Namespace Icm.CommandLineTools
    ''' <summary>
    '''  Subargument of a command line option.
    ''' </summary>
    ''' <remarks></remarks>

    Public Class SubArgument

        Private ReadOnly _type As SubArgumentType
        Private ReadOnly _description As String
        Private ReadOnly _name As String

        ReadOnly Property Name() As String
            Get
                Return _name
            End Get
        End Property

        ReadOnly Property Description() As String
            Get
                Return _description
            End Get
        End Property

        ReadOnly Property Type() As SubArgumentType
            Get
                Return _type
            End Get
        End Property

        Private Sub New(ByVal sName As String, eType As SubArgumentType)
            MyClass.New(sName, "", eType)
        End Sub

        Private Sub New(ByVal sName As String, ByVal sDescription As String, eType As SubArgumentType)
            _name = sName
            _description = sDescription
            _type = eType
        End Sub

        Public Shared Function List(ByVal sName As String) As SubArgument
            Return New SubArgument(sName, SubArgumentType.List)
        End Function

        Public Shared Function [Optional](ByVal sName As String) As SubArgument
            Return New SubArgument(sName, SubArgumentType.Optional)
        End Function

        Public Shared Function Required(ByVal sName As String) As SubArgument
            Return New SubArgument(sName, SubArgumentType.Required)
        End Function

        Public Shared Function List(ByVal sName As String, ByVal sDescription As String) As SubArgument
            Return New SubArgument(sName, sDescription, SubArgumentType.List)
        End Function

        Public Shared Function [Optional](ByVal sName As String, ByVal sDescription As String) As SubArgument
            Return New SubArgument(sName, sDescription, SubArgumentType.Optional)
        End Function

        Public Shared Function Required(ByVal sName As String, ByVal sDescription As String) As SubArgument
            Return New SubArgument(sName, sDescription, SubArgumentType.Required)
        End Function

        Public Function IsList() As Boolean
            Return Type = SubArgumentType.List
        End Function

        Public Function IsOptional() As Boolean
            Return Type = SubArgumentType.Optional
        End Function

        Public Function IsRequired() As Boolean
            Return Type = SubArgumentType.Required
        End Function
    End Class

End Namespace
